app/questionnaire/questionnaire_schema.py

```diff
@@ -1407,51 +1345,28 @@ def _get_rules_section_dependencies(
         )
 
     def _populate_calculated_summary_section_dependencies(self) -> None:
+        """
+        For each block, find all the calculated summary value source dependencies
+        and make sure all involved sections are added as dependencies for that block and its section.
+
+        Since calculated summaries can only contain answers from the section they are in,
+        it is sufficient to check the section for the calculated summary only.
+        """
         for section in self.get_sections():
             for block in self.get_blocks_for_section(section):
                 sources = get_mappings_with_key("source", block, ignore_keys=["when"])
-
-                calculated_summary_sources = [
-                    source
-                    for source in sources
-                    if source["source"] == "calculated_summary"
-                ]
-
-                section_dependencies = (
-                    self._get_calculated_summary_section_dependencies(
-                        sources=calculated_summary_sources,
-                    )
-                )
+                section_dependencies: set[str] = set()
+                for source in sources:
+                    if source["source"] == "calculated_summary":
+                        section_dependencies.add(
+                            # Type ignore: Validator ensures a valid identifier so section will exist
+                            self.get_section_id_for_block_id(source["identifier"])  # type: ignore
+                        )
```

Comment:
Minor, any reason for the formatting change?. Previous one was done to reduce nesting.

Reply:
I made the changes as I thought the extra function made it harder to follow and could be inlined
That said, this can be nested less. I've removed it by doing a comprehension instead.